### PR TITLE
Set up repo, loading of dataset and scaffolding for correction model

### PR DESCRIPTION
**Pull request by anton164**
**NB:** put val.jsonl and train.jsonl from [this google drive](https://drive.google.com/drive/folders/18Eqfemxf6wOQeSUNrZMlacF2OvwaRQ87) into `/data/paper`

## Key parts of this PR
- `train_correction_model.ipynb`demonstrates loading of data and training a correction model on a subset of this data
- `preprocessing/prepare_train_dataset.py` loads and tokenizes a dataset for training
- `model/correction_model.py` implements the scaffolding for the correction model. It includes a wrapper for training the model (fine-tuning BART for 2-label classification) & correcting generated summaries.

### CODE REVIEW

File: model/correction_model.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            self.model.parameters(), 
+            lr=learning_rate
+        )
+
+        self.model.train()
+        print("Training...")
+        for epoch in range(epochs):
+            start_time = time.time()
+            epoch_loss = 0
+            for pairs in tqdm(tokenized_dataset):
```
Comment (anton164) [inline model/correction_model.py:53]: For now I didn't implement this with proper batching

File: preprocessing/quantity_util.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# From https://github.com/CogComp/faithful_summarization
```
Comment (anton164) [inline preprocessing/quantity_util.py:1]: no need to review this, it's a copy-paste

File: preprocessing/make_entity_perturbations.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
+# From https://github.com/CogComp/faithful_summarization
```
Comment (anton164) [inline preprocessing/make_entity_perturbations.py:1]: no need to review this, it's a copy-paste
